src/include/simeng/arch/riscv/Instruction.hh

```diff
@@ -0,0 +1,261 @@
+              const InstructionMetadata& metadata, uint8_t latency,
+              uint8_t stallCycles);
+
+  /** Construct an instruction instance that raises an exception. */
+  Instruction(const Architecture& architecture,
+              const InstructionMetadata& metadata,
+              InstructionException exception);
+
+  /** Retrieve the identifier for the first exception that occurred during
+   * processing this instruction. */
+  virtual InstructionException getException() const;
+
+  /** Retrieve the source registers this instruction reads. */
+  const span<Register> getOperandRegisters() const override;
+
+  /** Retrieve the destination registers this instruction will write to.
+   * A register value of -1 signifies a Zero Register read, and should not be
+   * renamed. */
+  const span<Register> getDestinationRegisters() const override;
+
+  /** Check whether the operand at index `i` has had a value supplied. */
+  bool isOperandReady(int index) const override;
+
+  /** Override the specified source register with a renamed physical register.
+   */
+  void renameSource(uint8_t i, Register renamed) override;
+
+  /** Override the specified destination register with a renamed physical
+   * register. */
+  void renameDestination(uint8_t i, Register renamed) override;
+
+  /** Provide a value for the operand at the specified index. */
+  virtual void supplyOperand(uint8_t i, const RegisterValue& value) override;
+
+  /** Check whether all operand values have been supplied, and the instruction
+   * is ready to execute. */
+  bool canExecute() const override;
+
+  /** Execute the instruction. */
+  void execute() override;
+
+  /** Retrieve register results. */
+  const span<RegisterValue> getResults() const override;
+
+  /** Generate memory addresses this instruction wishes to access. */
+  span<const MemoryAccessTarget> generateAddresses() override;
+
+  /** Retrieve previously generated memory addresses. */
+  span<const MemoryAccessTarget> getGeneratedAddresses() const override;
+
+  /** Provide data from a requested memory address. */
+  void supplyData(uint64_t address, const RegisterValue& data) override;
+
+  /** Retrieve supplied memory data. */
+  span<const RegisterValue> getData() const override;
+
+  /** Early misprediction check; see if it's possible to determine whether the
+   * next instruction address was mispredicted without executing the
+   * instruction. */
+  std::tuple<bool, uint64_t> checkEarlyBranchMisprediction() const override;
+
+  /** Is this a store operation? */
+  bool isStore() const override;
+
+  /** Is this a load operation? */
+  bool isLoad() const override;
+
+  /** Is this a branch operation? */
+  bool isBranch() const override;
+
+  /** Is this a return instruction? */
+  bool isRET() const override;
+
+  /** Is this a branch and link instruction? */
+  bool isBL() const override;
+
+  /** Is this a SVE instruction? */
+  bool isSVE() const override;
```

Comment:
Don't think RISC-V has SVE support

Reply:
It's a pure virtual function so he has to override it, right?

Reply:
Yeah, have to override